src/function/vector_list_operation.cpp

```diff
@@ -320,5 +322,164 @@ std::unique_ptr<FunctionBindData> ListSliceVectorOperation::bindFunc(
     return std::make_unique<FunctionBindData>(arguments[0]->getDataType());
 }
 
+std::vector<std::unique_ptr<VectorOperationDefinition>> ListSortVectorOperation::getDefinitions() {
+    std::vector<std::unique_ptr<VectorOperationDefinition>> result;
+    result.push_back(std::make_unique<VectorOperationDefinition>(LIST_SORT_FUNC_NAME,
+        std::vector<DataTypeID>{VAR_LIST}, VAR_LIST, nullptr, nullptr, unaryBindFunc,
+        false /* isVarlength*/));
+    result.push_back(std::make_unique<VectorOperationDefinition>(LIST_SORT_FUNC_NAME,
+        std::vector<DataTypeID>{VAR_LIST, STRING}, VAR_LIST, nullptr, nullptr, binaryBindFunc,
+        false /* isVarlength*/));
+    result.push_back(std::make_unique<VectorOperationDefinition>(LIST_SORT_FUNC_NAME,
+        std::vector<DataTypeID>{VAR_LIST, STRING, STRING}, VAR_LIST, nullptr, nullptr,
+        ternaryBindFunc, false /* isVarlength*/));
+    return result;
+}
+
+std::unique_ptr<FunctionBindData> ListSortVectorOperation::unaryBindFunc(
+    const binder::expression_vector& arguments, FunctionDefinition* definition) {
```

Comment:
Instead of defining unary/binary/ternary functions for list sort operations. You can template the ascOrder/nullFirst parameters:
```
template <bool ascOrder = true, bool nullFirst = false>
static void sortValues(common::list_entry_t& input, common::list_entry_t& result,
        common::ValueVector& inputVector, common::ValueVector& resultVector)
```
When calling them:
```
sortValues<false, true>(input, result, inputVector, resultVector);  // ascOrder = false, nullFirst = true
```
The isAscOrder/nullFirst are constants at compile time, which means your binary/tenary bindFunc can reuse the unaryBindFunc. Also, you don't need to write three definitions for sortValue operations.